Application/Helper/View.hs

```diff
@@ -1,5 +1,40 @@
 module Application.Helper.View where
 
+import Generated.Types
 import IHP.ViewPrelude
+import Web.Types
 
--- Here you can add functions which are available in all your views
\ No newline at end of file
+-- Here you can add functions which are available in all your views
+renderExerciseWithMuscleGroupsForm :: [MuscleGroup] -> ExerciseWithMuscleGroups -> Html
+renderExerciseWithMuscleGroupsForm allMuscleGroups exerciseWithMuscleGroups =
+  formFor
+    exercise
+    [hsx|
+    {textField #name}
+    {renderMuscleGroupCheckboxGroup allMuscleGroups muscleGroups}
+    {submitButton}
+
+|]
+  where
+    ExerciseWithMuscleGroups {exercise, muscleGroups} = exerciseWithMuscleGroups
+    renderMuscleGroupCheckboxGroup :: [MuscleGroup] -> [MuscleGroup] -> Html
+    renderMuscleGroupCheckboxGroup allMuscleGroups exerciseMuscleGroups =
+      [hsx|
+        <fieldset>
+          <legend>Muscle Groups</legend>
+          {forEachWithIndex allMuscleGroups (renderMuscleGroupCheckbox exerciseMuscleGroups)}
+        </fieldset>
+      |]
+
+    renderMuscleGroupCheckbox :: [MuscleGroup] -> (Int, MuscleGroup) -> Html
+    renderMuscleGroupCheckbox selectedMuscleGroups (index, muscleGroup) =
+      let checkboxId = "checkbox-muscleGroup-" <> show index
+          checked = muscleGroup `elem` selectedMuscleGroups
+       in [hsx|
+          <div class="form-check">
+            <!-- Would be super nice if I could just use the checkboxField here -->
+            <!-- {checkboxField #muscleGroupIds} {fieldName = "muscleGroupIds"} -->
```

Comment:
Yeah it sucks that this does't work, hopefully we can fix this in the next major version of IHP. I'll add this to the roadmap